doc/source/whatsnew/v1.3.0.rst

```diff
@@ -893,6 +893,7 @@ Other
 - Bug in :meth:`DataFrame.equals`, :meth:`Series.equals`, :meth:`Index.equals` with object-dtype containing ``np.datetime64("NaT")`` or ``np.timedelta64("NaT")`` (:issue:`39650`)
 - Bug in :func:`pandas.util.show_versions` where console JSON output was not proper JSON (:issue:`39701`)
 - Bug in :meth:`DataFrame.convert_dtypes` incorrectly raised ValueError when called on an empty DataFrame (:issue:`40393`)
+- Bug in :meth:`DataFrame.agg()` sometimes not sorting the non-concatenated axis in the order of the provided aggragation functions.
```

Comment:
can you be more specific here (e.g. when instead of sometimes)

Reply:
Adjusted!